fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingThrottle.java

```diff
@@ -153,8 +158,9 @@ void decreaseLimit(@Nonnull FDBException fdbException,
                            @Nullable List<Object> additionalLogMessageKeyValues) {
             // TODO: decrease the limit only for certain errors
             countFailedTransactions++;
+            consecutiveFailureCount++;
             long oldLimit = recordsLimit;
-            recordsLimit = Math.max(1, Math.min(lastFailureRecordsScanned - 1, ((lastFailureRecordsScanned * 9) / 10)));
+            recordsLimit = Math.max(1, Math.min(lastFailureRecordsScanned - 1, ((lastFailureRecordsScanned * oneToNineFactor(consecutiveFailureCount)) / 10)));
```

Comment:
This approach doesn't work around the retries by the underlying runner, correct? So if the runner is set to retry 10 times, you'll still do 10 retries before decreasing the limit. Is that correct?

Reply:
Yes.